solc/CommandLineParser.cpp

```diff
@@ -1228,8 +1228,8 @@ void CommandLineParser::processArgs()
 	if (m_args.count(g_strYulOptimizations))
 	{
 		OptimiserSettings optimiserSettings = m_options.optimiserSettings();
-		if (!optimiserSettings.runYulOptimiser)
-			solThrow(CommandLineValidationError, "--" + g_strYulOptimizations + " is invalid if Yul optimizer is disabled");
+		if (!optimiserSettings.runYulOptimiser && !isEmptyOptimizerSequence(m_args[g_strYulOptimizations].as<std::string>()))
+			solThrow(CommandLineValidationError, "--" + g_strYulOptimizations + " is invalid with any sequence other than empty (:) if Yul optimizer is disabled.");
```

Comment:
I think that this will not clarify things for someone who does not know that `":"` represents an empty sequence (`(:)` looks confusing). And for those who know, this is superfluous so I'd keep it simple:

```suggestion
			solThrow(CommandLineValidationError, "--" + g_strYulOptimizations + " is invalid with a non-empty sequence if Yul optimizer is disabled.");
```

Reply:
I mean, if someone is directly passing a sequence of optimizer steps via `--yul-optimizations`, I would assume that they know what an empty sequence is, or think they know, which brings me to the second point - people may still be confused between `""` and `":"`, which is why I explicitly stated the empty sequence.

Reply:
That's an optimistic assumption.

I mean, my main problem with this is that `(:)` in the message looks odd to me. It just does not look like a sequence. I'm not sure if I would realize what you mean if I didn't know already. I originally wanted to suggest to use `(":")` or `(--yul-optimizations=":")` to make it clearer but in the end I thought it's better not to overthink this :)